lib/rules/no-uppercase-with-underscore-property-name.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { docUrl } = require('../util/doc-url');
+
+const INVALID_PROPERTY_NAME_REGEX = /(?=.*_).*[A-Z]/;
```

Comment:
As mentioned below, matching the property name prefix and reporting it would make the error message less clear. In this regard, I would suggest simplifying the regex and removing the lookahead.